packages/block-editor/src/store/reducer.js

```diff
@@ -1842,6 +1842,55 @@ export function temporarilyEditingAsBlocks( state = '', action ) {
 	return state;
 }
 
+/**
+ * Reducer returning the current configuration for block inspector tabs.
+ *
+ * @param {Object} state  Current configuration state for block inspector tabs.
+ * @param {Object} action Dispatched action.
+ *
+ * @return {Object} Updated state.
+ */
+export function blockInspectorTabs( state = {}, action ) {
```

Comment:
It's an interesting one, because reducers are usually used for values that are likely to change at runtime. I wonder if there will be a use case for changing these values while the editor is running.

My feeling is that these values will probably be set as an initial configuration of the editor, which is often the job of the block editor settings (although technically they can change at runtime too, but often don't).

I think the only pain point will be that it's a little more difficult to handle deeply nested objects, but I think the data structure could be simplified (as mentioned in another comment).

Editor settings can also be set in PHP, so that another advantage.

Reply:
Just to note we have a use case for the Nav offcanvas experiment whereby we'd need to programmatically select the default tab as a result of an interaction with the "edit" button.

Reply:
Appreciate the feedback 👍 

> because reducers are usually used for values that are likely to change at runtime

The desire to set the default tab for the Navigation Link block, depending on if it was edited via the parent Navigation block's list view, drove the need for these values to change at runtime.

> My feeling is that these values will probably be set as an initial configuration of the editor, which is often the job of the block editor settings 

Splitting the management of the block inspector tab display so the default tabs are handled separately would open up the possibility of using the block editor settings. I'll be happy to explore that approach as well.

In the meantime, I wonder if this is reason enough to reconsider disabling tabs entirely for the Navigation Link block in the short term. It would allow us to avoid adding anything to the store related to the tabs or navigation experiments and all tab management settings would still be contained in a single location.

Reply:
> Just to note we have a use case for the Nav offcanvas experiment whereby we'd need to programmatically select the default tab as a result of an interaction with the "edit" button.

@getdave Aaron explained this to me in a private message. My mistake for not reading the PR description thoroughly.

The main problem is that I don't think a stable API should be introduced for two experimental features, so that's the main issue for me. There is a proposal for experimental selectors/actions, which would help 
 - https://github.com/WordPress/gutenberg/pull/44521.
 
 I don't know if the `__experimental` prefix is still allowed? cc @adamziel 
 
Alternatively the prefix could be used, but the selectors/actions would need to be stabilized or removed for alternatives before 6.2, so that they're not released with WordPress core.

Reply:
> In the meantime, I wonder if this is reason enough to reconsider disabling tabs entirely for the Navigation Link block in the short term. It would allow us to avoid adding anything to the store related to the tabs or navigation experiments and all tab management settings would still be contained in a single location.

@scruffian If we go down this route then will this compromise the user testing of the Nav offcanvas experiment?

Reply:
I think as long as we can put the link settings above the color ones then this should be an ok interim solution.

Reply:
Here is what the navigation link block currently looks like with the tabs disabled. The label input is the first much like I imagine is desired.

<img width="250" alt="Screenshot 2022-12-05 at 6 19 36 pm" src="https://user-images.githubusercontent.com/60436221/205613346-39371d45-398a-468f-a99a-ec185ceae48f.png">

Reply:
> I don't know if the __experimental prefix is still allowed? cc @adamziel

Ideally not in anything exported. Once the tooling is merged it will be safe to stop introducing new prefixed APIs entirely, but we're not there yet.